integration/apidefinition_test.go

```diff
@@ -770,45 +771,230 @@ func TestApiDefinitionUpstreamCertificates(t *testing.T) {
+					reqContext := pkgclient.SetContext(context.Background(), pkgContext)
+					exists := klient.Universal.Certificate().Exists(reqContext, calculatedCertID)
+
+					is.True(exists)
+
+					return true, nil
+				}, wait.WithTimeout(defaultWaitTimeout), wait.WithInterval(defaultWaitInterval))
+				is.NoErr(err)
+
+				return ctx
+			}).Feature()
+
+	testenv.Test(t, adCreate)
+}
+
+func TestApiDefinitionClientMTLS(t *testing.T) {
+	var (
+		apiDefClientMTLSWithCert    = "apidef-client-mtls-with-cert"
+		apiDefClientMTLSWithoutCert = "apidef-client-mtls-without-cert"
+		defaultVersion              = "Default"
+		opNs                        = "tyk-operator-system"
+		certName                    = "test-tls-secret-name"
+		tykConnectionURL            = ""
+	)
+
+	mode := os.Getenv("TYK_MODE")
+
+	switch mode {
+	case "pro":
+		tykConnectionURL = adminLocalhost
+	case "ce":
+		tykConnectionURL = gatewayLocalhost
+	}
+
+	testWithCert := features.New("Client MTLS with certificate").
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string) //nolint: errcheck
+			is := is.New(t)
+			t.Log(testNS)
+
+			_, err := createTestTlsSecret(ctx, testNS, nil, envConf)
+			is.NoErr(err)
+
+			return ctx
+		}).
+		Setup(func(ctx context.Context, t *testing.T, envConf *envconf.Config) context.Context {
+			testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+			is := is.New(t)
+
+			// Create ApiDefinition with Client certificate
+			_, err := createTestAPIDef(ctx, testNS, func(apiDef *v1alpha1.ApiDefinition) {
+				apiDef.Name = apiDefClientMTLSWithCert
+				apiDef.Spec.UseMutualTLSAuth = true
+				apiDef.Spec.ClientCertificateRefs = []string{certName}
+				apiDef.Spec.VersionData.DefaultVersion = defaultVersion
+				apiDef.Spec.VersionData.NotVersioned = true
+				apiDef.Spec.VersionData.Versions = map[string]model.VersionInfo{
+					defaultVersion: {Name: defaultVersion},
+				}
+			}, envConf)
+			is.NoErr(err) // failed to create apiDefinition
+
+			return ctx
+		}).
+		Assess("API must have client certificate field defined",
+			func(ctx context.Context, t *testing.T, cfg *envconf.Config) context.Context {
+				is := is.New(t)
+				client := cfg.Client()
+				testNS := ctx.Value(ctxNSKey).(string) //nolint:errcheck
+
+				tlsSecret := v1.Secret{} //nolint:errcheck
+
+				err := client.Resources(testNS).Get(ctx, certName, testNS, &tlsSecret)
+				is.NoErr(err)
+
+				certPemBytes, ok := tlsSecret.Data["tls.crt"]
+				if !ok {
+					is.Fail()
+				}
+				certFingerPrint := cert.CalculateFingerPrint(certPemBytes)
+
+				opConfSecret := v1.Secret{}
+				err = client.Resources(opNs).Get(ctx, "tyk-operator-conf", opNs, &opConfSecret)
+				is.NoErr(err)
+
+				tykAuth, ok := opConfSecret.Data["TYK_AUTH"]
+				if !ok {
+					is.Fail()
+				}
+
+				tykOrg, ok := opConfSecret.Data["TYK_ORG"]
+				if !ok {
+					is.Fail()
+				}
+
+				if mode == "ce" {
+					tykOrg = []byte("")
+				}
```

Comment:
I am not sure, but is it true to set `tykOrg` to an empty byte if mode equals `"ce"`? Shouldn't it be the default value for CE mode? So, if `tykOrg` does not equal `[]byte("")` in CE, shouldn't that indicate that `opConfSecret` is invalid?

Reply:
We were setting to myOrg by default. I changed default value. I can now remove this.
Thanks for spotting it.